File: solutions/prolog/raindrops/2/raindrops.pl
convert(N, S) :-
    pling(N, PlingS),
    plang(N, PlangS),
    plong(N, PlongS),
    string_concat(PlingS, PlangS, S1),
    string_concat(S1, PlongS, S2),
    (S2 == "" -> number_string(N, S); S = S2).

pling(N, "Pling") :-
    0 is (N mod 3), !.
pling(_, "").

plang(N, "Plang") :-
    0 is (N mod 5), !.
plang(_, "").

plong(N, "Plong") :-
    0 is (N mod 7), !.
plong(_, "").

File: solutions/fortran/raindrops/1/raindrops.f90
module raindrops
   implicit none
contains

   function convert(i)
      integer, intent(in) :: i
      character(20) :: convert
      convert = ""
      if (mod(i, 3) == 0) then
         convert = "Pling"
      end if
      if (mod(i, 5) == 0) then
         convert = trim(convert)//"Plang"
      end if
      if (mod(i, 7) == 0) then
         convert = trim(convert)//"Plong"
      end if
      if (trim(convert) == "") then
         write (convert, '(G0)') i
      end if
   end function convert

end module raindrops
